src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -178,49 +260,52 @@ public static void runIndexingBenchmarks(List<IndexBenchmark> indexBenchmarks, S
 		}
 	}
 
-    private static Supplier<ControlledExecutor.CallableWithType<String, NamedList<Object>>> getQuerySupplier(QueryGenerator queryGenerator, HttpSolrClient client, String collection) {
-		class QueryCallable implements ControlledExecutor.CallableWithType<String, NamedList<Object>> {
-			private final QueryRequest queryRequest;
+	public static class QueryCallable implements ControlledExecutor.CallableWithType<String, SolrBenchQueryResponse> {
+		private final QueryRequest queryRequest;
+		private final String queryString;
+		private final HttpSolrClient client;
+		private final String collection;
+		
+		public QueryCallable(QueryRequest queryRequest, String queryString, HttpSolrClient client, String collection) {
+			this.queryRequest = queryRequest;
+			this.client = client;
+			this.collection = collection;
+			this.queryString = queryString;
+		}
+		@Override
+		public String getType() {
+			return queryRequest.toString();
+		}
 
-			private QueryCallable(QueryRequest queryRequest) {
-				this.queryRequest = queryRequest;
-			}
-			@Override
-			public String getType() {
-				return queryRequest.toString();
-			}
+		public QueryRequest getQueryRequest() {
+			return queryRequest;
+		}
 
-			@Override
-			public NamedList<Object> call() throws Exception {
-				NamedList<Object> rsp = client.request(queryRequest, collection);
-				//let's not do printErrOutput here as this reads the input stream and once read it cannot be read anymore
-				//Probably better to let the caller handle the return values instead
-				//printErrOutput(queryRequest, rsp);
-				return rsp;
-			}
+		public String getQueryString() {
+			return queryString;
 		}
+
+		@Override
+		public SolrBenchQueryResponse call() throws Exception {
+			NamedList<Object> rsp = client.request(queryRequest, collection);
+			//let's not do printErrOutput here as this reads the input stream and once read it cannot be read anymore
+			//Probably better to let the caller handle the return values instead
+			//printErrOutput(queryRequest, rsp);
+			return new SolrBenchQueryResponse(queryString, rsp);
```

Comment:
We might not need this wrapper if queryString is the same as `getType()`, since the listener invocation ` executionListener.onExecutionComplete(type, result, durationInNanoSec);`  does get the `type`, which is the query.